mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/impl/MongoDBIndexOptions.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.options.impl;
+
+
+/**
+ * MongoDB specific options for a given index.
+ *
+ * See <a href="https://docs.mongodb.com/manual/reference/method/db.collection.createIndex/#options-for-all-index-types">https://docs.mongodb.com/manual/reference/method/db.collection.createIndex/#options-for-all-index-types</a>
+ *
+ * @author Guillaume Smet
+ */
+public class MongoDBIndexOptions {
+
+	/**
+	 * The target index name.
+	 */
+	private String targetIndexName;
+
+	/**
+	 * Builds the index in the background so that building an index does not block other database activities.
+	 */
+	private boolean background;
+
+	/**
+	 * Allows to define a partial index by providing a filter expression.
+	 *
+	 * Must be a valid MongoDB document.
+	 */
+	private String partialFilterExpression;
+
+	/**
+	 * If true, the index only references documents with the specified field.
+	 * These indexes use less space but behave differently in some situations (particularly sorts).
+	 * The default value is false
+	 */
+	private boolean sparse;
+
+	/**
+	 * Specifies a value, in seconds, as a TTL to control
+	 * how long MongoDB retains documents in this collection
+	 */
+	private int expireAfterSeconds;
+
+	/**
+	 * Allows to define options for the storage engine.
+	 *
+	 * Must be a valid MongoDB document.
```

Comment:
No. It has to be a valid MongoDB document e.g. ~ a JSON document (but with some extension). It also has to be valid as a storage engine option document but this cannot be checked by us.